addons/AutoExportVersion/AutoExportVersion.gd

```diff
@@ -1,147 +1,185 @@
+
+
+
+####################################################################################################
+#################################  Unimportant plugin stuff here  ##################################
+####################################################################################################
+
+
+const _CURRENT_VERSION: String = "Current version: {version}"
+const _EMPTY_VERSION_ERROR: String = "Version string is empty.\nMake sure your 'get_version()' in '{script_path}' is configured properly."
+
+const _TOOL_MENU_ITEM_NAME: String = "AutoExport: Print Current Version"
+
+var _exporter: AutoExportVersionExporter
 
-var exporter: AEVExporter
 
 func _enter_tree() -> void:
-	exporter = AEVExporter.new()
-	exporter.plugin = self
-	add_export_plugin(exporter)
-	add_tool_menu_item("Print Current Version", self, "print_version")
-	
-	if not File.new().file_exists(VERSION_SCRIPT_PATH):
-		exporter.store_version(_fetch_version(PoolStringArray(), true, "", 0))
+	_exporter = AutoExportVersionExporter.new()
+	_exporter.plugin = self
+	add_export_plugin(_exporter)
+	add_tool_menu_item(_TOOL_MENU_ITEM_NAME, _tool_menu_print_version)
 
 func _exit_tree() -> void:
-	remove_export_plugin(exporter)
-	remove_tool_menu_item("Print Current Version")
-
-func print_version(ud):
-	var v = _fetch_version(PoolStringArray(), true, "", 0)
-	if v.empty():
-		OS.alert("Error fetching version. Check console for details.")
-	else:
-		OS.alert("Current game version: %s" % v)
-		print(v)
-
-class AEVExporter extends EditorExportPlugin:
-	var plugin
+	remove_export_plugin(_exporter)
+	remove_tool_menu_item(_TOOL_MENU_ITEM_NAME)
+
+func _tool_menu_print_version() -> void:
+	var version: String = get_version(PackedStringArray(), true, "", 0)
 	
-	func _export_begin(features: PoolStringArray, is_debug: bool, path: String, flags: int):
-		var version: String = plugin._fetch_version(features, is_debug, path, flags)
-		if version.empty():
-			push_error("Version string is empty. Make sure your _fetch_version() is configured properly.")
-		
-		store_version(version)
+	if version.is_empty():
+		printerr(_EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		OS.alert(_EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		return
+	
+	printerr(_CURRENT_VERSION.format({"version":version}))
+	OS.alert(_CURRENT_VERSION.format({"version":version}))
+
 
-	func store_version(version: String):
-		var script = GDScript.new()
-		script.source_code = str("extends Reference\nconst VERSION = \"", version, "\"\n")
-		if ResourceSaver.save(VERSION_SCRIPT_PATH, script) != OK:
-			push_error("Failed to save version file. Make sure the path is valid.")
+class AutoExportVersionExporter extends EditorExportPlugin:
+	var plugin: EditorPlugin
+	
+	func _export_begin(features: PackedStringArray, is_debug: bool, path: String, flags: int) -> void:
+		if not plugin:
+			push_error("No plugin set in AutoExportVersionExporter")
```

Comment:
There was an error about empty version here. I think it should be re-added.

Also the new error is not necessary; it can only appear if someone meddles with code they shouldn't touch.

Reply:
The error about empty version is now handled in the `store_version()` function. 

Altrough it should be handled by the individual store location functions, if someone wants to use them directly.